src/applications/personalization/profile360/containers/CopyMailingAddress.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import isEqual from 'lodash/isEqual';
+import pick from 'lodash/pick';
+
+import {
+  consolidateAddress
+} from '../../../../platform/forms/address/helpers';
+
+import {
+  FIELD_NAMES
+} from '../constants/vet360';
+
+import {
+  selectVet360Field,
+  selectEditedFormField
+} from '../selectors';
+
+import {
+  cleanAddressDataForUpdate
+} from '../actions/misc';
+
+class CopyMailingAddress extends React.Component {
+  onChange = (event) => {
+    event.stopPropagation();
+    if (event.target.checked) {
+      this.props.copyMailingAddress(
+        consolidateAddress(this.props.mailingAddress)
+      );
+    }
+  }
+
+  render() {
+    if (this.props.hasEmptyMailingAddress) return <div/>;
+    return (
+      <div className="copy-mailing-address-to-residential-address">
+        <div className="form-checkbox-buttons">
+          <input
+            type="checkbox"
+            name="copy-mailing-address-to-residential-address"
+            id="copy-mailing-address-to-residential-address"
+            autoComplete="false"
+            checked={this.props.checked}
+            onChange={this.onChange}/>
+          <label htmlFor="copy-mailing-address-to-residential-address">
+            My home address is the same as my mailing address.
+          </label>
+        </div>
+      </div>
+    );
+  }
+}
+
+function mapStateToProps(state) {
+  const addressProps = [
+    'addressLine1',
+    'addressLine2',
+    'addressLine3',
+    'addressPou',
+    'addressType',
+    'city',
+    'countryName',
+    'internationalPostalCode',
+    'province',
+    'stateCode',
+    'zipCode'
+  ];
+
+  const mailingAddress = selectVet360Field(state, FIELD_NAMES.MAILING_ADDRESS);
+  const hasEmptyMailingAddress = !mailingAddress;
+
+  const residentialAddress = selectEditedFormField(state, FIELD_NAMES.RESIDENTIAL_ADDRESS).value;
+
+  const mailingAddressConsolidated = pick(cleanAddressDataForUpdate(consolidateAddress(mailingAddress)), addressProps);
+  const residentialAddressConsolidated = pick(consolidateAddress(residentialAddress), addressProps);
```

Comment:
@ncksllvn I think `consolidateAddress` is a vestige of PICU functionality; when we submit to Vet360, attributes like `militaryPostOfficeTypeCode` and `militaryStateCode` are automatically stripped away already. Why is `cleanAddressDataForUpdate` only called for `mailingAddress` and not also for `residentialAddress`?

Reply:
Following the transformations every step was important because the checkbox needed to stay in sync at all times (even if they copy over the mailing address, they could still change a value), and it was tricky trying to apply transformations on edited form values to compare with the API's value for the other address.